var/spack/repos/builtin/packages/petsc/package.py

```diff
@@ -1,40 +1,86 @@
+import os
 from spack import *
 
+
 class Petsc(Package):
-    """PETSc is a suite of data structures and routines for the
-       scalable (parallel) solution of scientific applications modeled by
-       partial differential equations."""
+    """
+    PETSc is a suite of data structures and routines for the scalable (parallel) solution of scientific applications
+    modeled by partial differential equations.
+    """
 
     homepage = "http://www.mcs.anl.gov/petsc/index.html"
-    url      = "http://ftp.mcs.anl.gov/pub/petsc/release-snapshots/petsc-3.5.3.tar.gz"
+    url = "http://ftp.mcs.anl.gov/pub/petsc/release-snapshots/petsc-3.5.3.tar.gz"
 
+    version('3.6.3', '91dd3522de5a5ef039ff8f50800db606')
     version('3.5.3', 'd4fd2734661e89f18ac6014b5dd1ef2f')
     version('3.5.2', 'ad170802b3b058b5deb9cd1f968e7e13')
     version('3.5.1', 'a557e029711ebf425544e117ffa44d8f')
 
-    depends_on("python @2.6:2.9")   # requires Python for building
+    variant('shared', default=True, description='Enables the build of shared libraries')
+    variant('mpi', default=True, description='Activates MPI support')
+    variant('double', default=True, description='Switches between single and double precision')
+
+    variant('metis', default=True, description='Activates support for metis and parmetis')
+    variant('hdf5', default=True, description='Activates support for HDF5 (only parallel)')
+    variant('boost', default=True, description='Activates support for Boost')
+    variant('hypre', default=True, description='Activates support for Hypre')
+
+    # Build dependencies
+    depends_on('python @2.6:2.9')  # requires Python for building
+
+    # Virtual dependencies
+    depends_on('blas')
+    depends_on('lapack')
+    depends_on('mpi', when='+mpi')
 
-    depends_on("boost")
-    depends_on("blas")
-    depends_on("lapack")
-    depends_on("hypre")
-    depends_on("parmetis")
-    depends_on("metis")
-    depends_on("hdf5+mpi")
-    depends_on("mpi")
+    # Other dependencies
+    depends_on('boost', when='+boost')
+    depends_on('metis', when='+metis')
+
+    depends_on('hdf5~cxx~unsupported+mpi', when='+hdf5+mpi')
```

Comment:
`unsupported` in HDF5's configurations doesn't mean that it enables dodgy code. It only enables additional configuration options, or rather combinations of configuration options. Thus, if something doesn't work with Intel, something else needs to be disabled, and `unsupported` can remain enabled.

Here, it seems that the Intel compiler can't handle HDF5's C++ interface.

Which version of the Intel compiler is this? What is the error message? What system is this?

If we disable something, we should keep a record of why this was done.